### ISSUE TITLE
Win32: localtime_r() shim compilation error

### ISSUE THREAD

**Issue by AustinDeng** [oatpp/oatpp#410]
As a beginner, I regret to find a problem but I can’t do anything to solve it.

I used the first example of this website[https://oatpp.io/docs/start/step-by-step/#api-low-level-components-overview] to run the program.

run in visual studio 2019 community and all configurations are default configurations.
    
    Microsoft (R) Windows (R) Resource Compiler Version 10.0.10011.16384

Unfortunately, the compilation failed.

it says:"Errno_t _localtime32_s(tm *,const __time32_t *)": Cannot convert parameter 2 from "time_t *" to "const __time32_t *"

The error is located under this file: oatpp\src\oatpp\core\base\Environment.cpp, line 46
The reason is that: errno_t _localtime32_s(tm *,const __time32_t *)": Cannot convert parameter 2 from "time_t *" to "const __time32_t *

``` C++
#elif (defined(WIN32) || defined(_WIN32)) && not defined(_WIN64)
struct tm* localtime_r(time_t *_clock, struct tm *_result) {
    _localtime32_s(_result, _clock);  // Here is line 46. the _clock is time_t*, but _localtime32_s need ‘const __time32_t*’
    return _result;
}
#endif
```

Here is _localtime32_s in time.h:
``` C++
_Check_return_wat_
_ACRTIMP errno_t __cdecl _localtime32_s(
    _Out_ struct tm*        _Tm,
    _In_  __time32_t const* _Time
    );
```

I tried to find a solution online, but none of them worked.

I would be very grateful if you could help.

**Comment by bamkrs** [comment]
Hey @AustinDeng are you compiling for 32 or 64 bit?

**Comment by AustinDeng** [comment]
@bamkrs  Thank you very much for your reply. I checked it and found that my compiler uses 32-bit compilation. After I changed it to 64-bit, the problem was solved. Thanks again.

**Comment by bamkrs** [comment]
I would like to reopen the issue since there is definitely a problem with Win32. We should look into that at some time.

**Comment by sgallou** [comment]
Seem to be fixed by 6ad5cd2791544e1224bf9ae44ffeb39a3807f4e7

**Comment by lganzzzo** [comment]
Nice!
I'm closing this issue for now.
Please feel free to reopen it if needed.
